### PR TITLE
add icon to search results label for item type

### LINKED ISSUE #715: Update list view of search results - A - Result item label
Split out of #669 

Screen Shot 2019-08-28 at 4 36 48 PM

**A: Result item label** - No change from current, except:

- [x] Prefix the label with an icon conveying whether the item is a collection, a file (lowest-level component), or something in between those two levels.

### PR DESCRIPTION
**Pull request by labradford**
Closes #715 part of #669 
### Issue
Prefix the label with an icon conveying whether the item is a collection, a file (lowest-level component), or something in between those two levels.

### Solution
- add helper method to determine the icon based on document['level_ssm']. There will be three options: Collection, File and all others
- add icon to _index_header.html.erb partial
- add styling
- add specs for helper and view

### Notes
- Currently uses placeholder icons
- Styling will need to be updated when the icons are provided

### Result
<a href="https://cl.ly/c858e5f4fd48" target="_blank"></a>

### CODE REVIEW

File: app/helpers/arclight_helper.rb [hunk 1]
```diff
@@ -113,6 +113,19 @@ def display_size(size)
     size.to_s
   end
 
+  # determine which icon to show in search results header
+  # these icon names will need to be updated when the icons are determined
+  def document_header_icon(document)
+    level = document['level_ssm'] || nil
```
Comment (mejackreed) [inline app/helpers/arclight_helper.rb]: Can we just use the accessor already available on `document`?

```ruby
document.level
```

Comment (labradford) [reply app/helpers/arclight_helper.rb]: updated to use document.level

File: app/helpers/arclight_helper.rb [hunk 2]
```diff
@@ -113,6 +113,19 @@ def display_size(size)
     size.to_s
   end
 
+  # determine which icon to show in search results header
+  # these icon names will need to be updated when the icons are determined
+  def document_header_icon(document)
+    level = document.level || nil
```
Comment (mejackreed) [inline app/helpers/arclight_helper.rb]: I don't think we need to do the `nil` check here anymore as the accessor in `SolrDocument` handles that.

What do you think about switching this to a case statement as opposed to the nested if/else?

Comment (labradford) [reply app/helpers/arclight_helper.rb]: Refactored to case statement.

**Comment by mejackreed** [comment]
@seanaery 

> Use something other than the magnifying glass as the collection icon -- that one is signaling it could be clicked on to power some kind of search functionality.

What would you suggest as a temporary fix? Something that is already available in ArcLight or Blacklight?

**Comment by labradford** [comment]
> Hi @labradford - I think this is a generally a good approach and using the `blacklight_icon` helper is a good way to render SVGs. Even though the icons here are temporary placeholders until the real ones get finalized, what would you think of making these revisions?:
> 
> 1. Use more than two different icons (add at least a third that distinguishes a File from Series/Subseries/etc).
> 2. Use something other than the magnifying glass as the collection icon -- that one is signaling it could be clicked on to power some kind of search functionality.

These were the only icons I could get to render, so this is why I chose them, knowing that we will be adding icons at a later time.

**Comment by seanaery** [comment]
@mejackreed @labradford - I see now there are really only a couple SVGs currently available via Blacklight/ArcLight. I was thinking it'd be easy to grab a free fontawesome SVG like https://fontawesome.com/icons/archive?style=solid and put in the ArcLight assets though looking closer there's a bit of attribution required; that may be unnecessarily complicated for something that will probably be short-lived in the UI.

**Pull request merged**